bldnrnmacpkg.sh

```diff
@@ -24,10 +34,15 @@ if test "$args" = "" ; then
   args="$default_pythons"
 fi
 
+MPI_LIST="/opt/homebrew/opt/openmpi/include;/opt/homebrew/opt/mpich/include"
```

Comment:
Does this mean that the user needs to have homebrew-installed versions of MPI installed on their machine? I.e. any other installed MPI (in other locations) won't work?

Note that the `/opt/homebrew` part is specific to arm64, on x86 it's `/usr/local` (obtainable with `brew --prefix`).

Reply:
This is not really a user ready file. It is just a script I use on my machine to create a universal2 mac pkg.
Perhaps someday we will be doing this when we do remote wheel building (the blocker is notarization). Good point about /opt/homebrew vs /usr/local . That, and probably quite a few other details, will have to be incorporated in a future development.

Reply:
> needs to have homebrew-installed versions of MPI installed on their machine?

I may have missed the import of your question. The user machine is completely agnostic about the location of MPI or even if it exists. That is because the mpi library is dynamically loaded after launching python or nrniv.
And then the proper nrnmpi library is loaded based on the mpi library version that got loaded.

Reply:
> the mpi library is dynamically loaded after launching python or nrniv

That's the part that I am curious about; does the location need to be fixed (namely, does it have to be in one of the locations specified in `MPI_LIST`), or can it be installed elsewhere? I'm guessing in case of a custom MPI install location, that the user can specify some env variable to force NEURON to look for MPI there?

Reply:
> can it be installed elsewhere? I'm guessing in case of a custom MPI install location, that the user can specify some env variable to force NEURON to look for MPI there?

The user can install MPI (either ompi or mpich) anywhere. NEURON has no idea where they are. If the user dynamic load system can't find libmpi, or whatever it's name is, then the user will have to set the environment variable MPI_LIB_NRN_PATH as the full path to libmpi...

Reply:
Makes sense, thanks for the clarification!